src/main/java/io/gingersnapproject/rest/EngineResource.java

```diff
@@ -2,23 +2,35 @@
 
 import javax.inject.Inject;
 import javax.ws.rs.GET;
+import javax.ws.rs.POST;
 import javax.ws.rs.Path;
 
-import io.gingersnapproject.cdc.ManagedEngine;
+import io.gingersnapproject.cdc.DynamicRuleManagement;
+import io.gingersnapproject.cdc.RuleLifecycleManagement;
+import io.gingersnapproject.cdc.configuration.DefaultRuleParser;
+
 import org.jboss.resteasy.reactive.RestQuery;
 
-@Path("/v1/engine")
+@Path("/v1/rules")
```

Comment:
Long term I don't think we need this endpoint for the Gingersnap use-case. That being said, I'm not against merging for now if it's useful for dev/debugging.

Is there a need for this for the Infinispan use-case? If so, in order for this to remain we would need to add authentication for it to be releasable, or even better allow it to be disabled via the config.

Reply:
I agree with it being available only for dev mode for now and we can remove it later. But I am not sure on the ISPN side TBH. @wburns, do you see something here?